code/script.js

```diff
@@ -33,13 +42,346 @@ const showMessage = (message, sender) => {
   chat.scrollTop = chat.scrollHeight;
 }
 
-// Starts here
+
+  // Starts here - The greetUser function that is called at the end and starts the chat bot running is defined here.
 const greetUser = () => {
   // here we call the function showMessage, that we declared earlier with the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, What's your name?", 'bot');
+  showMessage("Hi there! I'm Crafty, the Idea Bot, who are you?", 'bot');
   // Just to check it out, change 'bot' to 'user' here 👆
 }
 
+
+//This makes the submit button clickable and makes the first bubble pop up.
+form.addEventListener('submit', (event) => {
+  event.preventDefault()
+  username = nameInput.value // isn't just defined locally b/c haven't reused "let", adding the declared value to the global "let username ="
+  showMessage(username, 'user')
+  setTimeout(() => showBotResponseOne(), 500) //calling showBotResponse 500ms after the user's name shows up
+});
+
+
+//This is where the showBotResponse function is declared - this will show the bot's response to the user's name.
+const showBotResponseOne = () => { 
+  if(username !== "") {
+    showMessage(`Hey, ${username}! Which craft do you want an idea for?`, 'bot')
+    input.innerHTML = `
+      <button id="quilt-button">Quilting</button>
+      <button id="crochet-button">Crochet</button>
+      <button id="cross-button">Cross-Stitch</button>
+    `
+    showCraftButtons(); //This function lives here instead of outside of this function because it ensures that the buttons exist in the HTML before they are called.
+  //                    //if this was placed outside of this function, it would be called immediately, the buttons wouldn't exist yet, and the code would break.
+
```

Comment:
Again, what a great way of working, taking notes next to the code which is where they matter the most. It seems so obvious now but really is a new way of working. I like that you declare why it can't be placed somewhere else, allowing us to understand your choices and letting us in on your line of thought.